### PR TITLE
Add sentiment analysis twitter blog post

### PR DESCRIPTION
**Pull request by feconroses**
## What does this PR do?
Adds a new blog post that provides a nice introduction to **_Sentiment Analysis with Twitter_**, for coders _and_ no coders 💥

Wrote this blog post to 1) promote the Inference API and 2) use SEO best practices for ranking in the top positions of Google for the high-volume keyword `sentiment analysis twitter`.

Hopefully, this post will help tons of people get started with Sentiment Analysis on Twitter, get them to use the Inference API, and generate more upgrades 🚀

Please let me know your thoughts and suggestions 🙏 

Btw, I revoked the Hugging Face token that is shown in some of the gifs.

## Screenshot
Getting Started with Sentiment Analysis on Twitter

### CODE REVIEW

**Comment by BritneyMuller** [comment]
Great job @feconroses !!! Made some very minor grammatical fixes. 

Don't forget periods when using emojis. 🎉 Have a hunch Google still evaluates the grammatical quality of pages. 🤗

**Review by BritneyMuller — APPROVED** [review]
Looks great!

**Comment by feconroses** [comment]
@BritneyMuller thank you so much! 🙌

**Comment by osanseviero** [comment]
The combined assets of this blog are around 20MB, which is too much for a website. It will make the experience very slow and/or expensive for mobile. I also see a file `assets/85_sentiment_analysis_twitter/sentiment-analysis-twitter.md` which I think should be removed

**Review by osanseviero — COMMENTED** [review]
We already have:
* https://huggingface.co/blog/sentiment-analysis-python
* https://huggingface.co/blog/opinion-classification-with-kili
* https://huggingface.co/blog/supercharge-customer-service-with-machine-learning

So three blog posts about sentiment analysis. I understand that the audiences might differ a bit, and that sentiment analysis is searched significantly, but I'm concerned of having so many blog posts of same topic. For next blog posts it might be more exciting to solve a different problem

File: sentiment-analysis-twitter.md [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+
+Nowadays, getting started with sentiment analysis on Twitter is quite easy and straightforward 🙌 
+
+With a few lines of code, you can automatically get tweets, run sentiment analysis and visualize the results. And you can learn how to do all these things in just a few minutes! 
+
+In this section, we'll show you how to do it with a cool little project: we'll do sentiment analysis of tweets mentioning [Notion](https://twitter.com/notionhq)! 
+
+First, you'll use [Tweepy](https://www.tweepy.org/), an open source Python library to get tweets mentioning @NotionHQ using the [Twitter API](https://developer.twitter.com/en/docs/twitter-api). Then you'll use the [Inference API](https://huggingface.co/inference-api) for doing sentiment analysis. Once you get the sentiment analysis results, you will create some charts to visualize the results and detect some interesting insights.
+
+You can use this [Google Colab notebook](https://colab.research.google.com/drive/1R92sbqKMI0QivJhHOp1T03UDaPUhhr6x?usp=sharing) to follow this tutorial.
```
Comment (osanseviero) [inline sentiment-analysis-twitter.md:81]: The colab can also be here in the /notebooks directory

File: sentiment-analysis-twitter.md [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+   response = requests.post(API_URL, headers=headers, json=payload)
+ return response.json()
+```
+
+Now, you are ready to do sentiment analysis on each tweet. 🔥🔥🔥
+
+```python
+tweets_analysis = []
+for tweet in tweets:
+   try:
```
Comment (osanseviero) [inline sentiment-analysis-twitter.md:186]: What kind of errors might you face here?

File: sentiment-analysis-twitter.md [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+```
+
+Next, you will create the API call using the `model id` and `hf_token`:
+
+```python
+API_URL = "https://api-inference.huggingface.co/models/" + model
+headers = {"Authorization": "Bearer %s" % (hf_token)}
+ 
+def analysis(payload):
+ response = requests.post(API_URL, headers=headers, json=payload)
```
Comment (osanseviero) [inline sentiment-analysis-twitter.md:173]: The indentation seems a bit inconsistent between sections

File: sentiment-analysis-twitter.md [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+ while "is currently loading" in str(response.json()): # retry request while model loads
```
Comment (osanseviero) [inline sentiment-analysis-twitter.md:174]: You can use `wait_for_model` in the parameters so you don't have to do a retry until the model loads.
https://huggingface.co/docs/api-inference/detailed_parameters

File: sentiment-analysis-twitter.md [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+  <figcaption>Word cloud for negative tweets</figcaption>
+</figure>
+
+That was fun, right? 
+
+With just a few lines of code, you were able to automatically gather tweets mentioning Notion using Tweepy, analyze them with a sentiment analysis model using the [Inference API](https://huggingface.co/inference-api), and finally create some visualizations to analyze the results. 💥 
+
+Are you interested in doing more? As a next step, you could use a second [text classifier](https://huggingface.co/tasks/text-classification) to classify each tweet by their theme or topic. This way, each tweet will be labeled with both sentiment and topic, and you can get more granular insights (e.g. are users praising how easy to use is Notion but are complaining about their pricing or customer support?).
+
+## How to do Twitter sentiment analysis without coding?
```
Comment (osanseviero) [inline sentiment-analysis-twitter.md:302]: I think this section should be linked near the beginning where you mention you also show how to do this without code

**Review by osanseviero — APPROVED** [review]
I did a light read but this is very cool!

File: sentiment-analysis-twitter.md (hunk 1 above)
Comment (feconroses) [reply sentiment-analysis-twitter.md:81]: Good idea, done! https://github.com/huggingface/blog/blob/main/notebooks/85_sentiment_analysis_twitter.ipynb

File: sentiment-analysis-twitter.md (hunk 4 above)
Comment (feconroses) [reply sentiment-analysis-twitter.md:174]: Oh great idea! I didn't know that option existed; it's much better than retrying until the model loads. Added it to the post.

File: sentiment-analysis-twitter.md (hunk 5 above)
Comment (feconroses) [reply sentiment-analysis-twitter.md:302]: It is linked at the beginning of the post in the table of contents (I prefer to not add an additional link in the introduction to the same section)

File: sentiment-analysis-twitter.md (hunk 2 above)
Comment (feconroses) [reply sentiment-analysis-twitter.md:186]: To be honest, I tend to do a "try" whenever I make a request to an API, as a few things usually could go wrong. In this case, for example, incorrectly adding the `hf_token` a few lines before this.

**Comment by feconroses** [comment]
@osanseviero thanks for taking the time to share your feedback! I really appreciate it 😃 
- Applied your inline suggestions!
- Removed the unnecessary markdown file on `assets/85_sentiment_analysis_twitter/`
- Removed some unused assets. Now, the combined assets of this blog post are around 5 MB. Honestly, I wouldn't worry much about this. At MonkeyLearn, we did heavier content with more GIFs, and those ranked well on Google (Google indexes mobile-first for a while now, and those contents still ranked at the top positions of Google). If you feel strongly about this, we can try to do lazy loading on these assets or use a better format for the videos that are not as heavy as GIFs (I tried WebP, but it doesn't seem much better for gifs).
- RE too much content about sentiment Analysis: we plan to cover more things from an SEO perspective! So far, two of the first three SEO-driven experiments were about sentiment analysis, as this is the most searched application in NLP on Google by a wide margin, so it's the easiest to show the impact we can have with SEO. But yes, over time, we want to publish SEO-driven content that can rank at top positions in Google for popular keywords related computer vision, audio, NLP, reinforcement learning, bert, etc.

**Comment by feconroses** [comment]
@osanseviero I'm not sure how I can resolve the existing conflicts; whenever I click on "Resolve conflicts", it takes me to this page, but it never finishes loading (the spinner keeps "spinning"):
Screen Shot 2022-07-04 at 4 48 48 PM

Do you know how to fix this?

**Comment by osanseviero** [comment]
It seems GH had some issues yesterday, it seems to be working fine now :)

**Comment by feconroses** [comment]
It was weird, Chrome was still stuck for me when trying to resolve the existing conflicts, but it worked as normal on Firefox 🥳

**Pull request merged**